src/utils/Api.js

```diff
@@ -0,0 +1,122 @@
+class Api {
+  constructor({ baseUrl, headers }) {
+    this._baseUrl = baseUrl;
+    this._headers = headers;
+  }
+
+  getAppInfo() {
+    return Promise.all([this.getInitialCards(), this.getUserInfo()]);
+  }
+
+  editUserInfo({ name, about }) {
+    return fetch(`${this._baseUrl}/users/me`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({ name, about }),
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error editing user info:", err);
+        throw err;
+      });
+  }
+
+  editAvatar({ avatar }) {
+    // debugger;
+    return fetch(`${this._baseUrl}/users/me/avatar`, {
+      method: "PATCH",
+      headers: this._headers,
+      body: JSON.stringify({ avatar }),
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error editing user info:", err);
+        throw err;
+      });
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers,
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error fetching user info:", err);
+        throw err;
+      });
+  }
+
+  getInitialCards() {
+    return fetch(`${this._baseUrl}/cards`, {
+      headers: this._headers,
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error fetching initial cards:", err);
+        throw err;
+      });
+  }
+
+  addNewCard({ name, link }) {
+    return fetch(`${this._baseUrl}/cards`, {
+      method: "POST",
+      headers: this._headers,
+      body: JSON.stringify({ name, link }),
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error adding card:", err);
+        throw err;
+      });
+  }
+
+  removeCard(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}`, {
+      method: "DELETE",
+      headers: this._headers,
+    })
+      .then((res) => this._checkResponse(res))
+      .catch((err) => {
+        console.error("Error deleting card:", err);
+        throw err;
+      });
+  }
+
+  addLike(cardId) {
+    return fetch(`${this._baseUrl}/cards/${cardId}/likes`, {
+      method: "PUT",
+      headers: this._headers,
+    })
+      .then((res) => {
+        if (res.ok) {
+          return res.json();
+        }
+        return Promise.reject(`Error: ${res.status}`);
```

Comment:
Please, reuse  ` .then((res) => this._checkResponse(res))` everywhere, to avoid code duplication

Reply:
* My remark hasn't been corrected https://github.com/TheKunFo/se_project_spots/pull/1#discussion_r2083013275

Please, reuse  ` .then((res) => this._checkResponse(res))` everywhere, to avoid code duplication